fetcher/fetchers.py

```diff
@@ -136,8 +136,14 @@ def is_exportable(self, obj):
             return False
         if obj.get("has_unpublished_ancestor"):
             return False
-        if obj.get("id_0") and not obj.get("id_0").startswith("FA"):
-            return False
+        if len(settings.ARCHIVESSPACE["resource_id_0_prefixes"]):
+            if obj.get("id_0") and not any(
+                    [obj.get("id_0").startswith(prefix) for prefix in settings.ARCHIVESSPACE["resource_id_0_prefixes"]]):
+                return False
+        if len(settings.ARCHIVESSPACE["finding_aid_status_restrict"]):
+            if not obj.get("finding_aid_status_ancestor") or any(
```

Comment:
The property `finding_aid_status_ancestor` doesn't actually exist yet, to my reading.

I think the definition for it might want to live here:
https://github.com/RockefellerArchiveCenter/pisces/blob/a2f91b8cd489b6dc264e1b1b73274d7263e40f0a/transformer/resources/source.py#L237-L247

But I'm not seeing how it would be populated (based on a parallel of "has_unpublished_ancestor").

Reply:
Yeah I just copied in the code from the issue, which was admittedly a lazy approach!

I think we're going to have to add some checking of the `jsonmodel_type` first, because this check really only applies to resources and archival objects. In the first case, we can just take the `finding_aid_status` from the resource itself; in the second we should have resolved ancestors, so we can take the last one of those (which will be the parent resource record) and use the `finding_aid_status` from there.

`is_exportable` is getting pretty complex and I'm wondering if it makes sense to refactor this a bit so that each of those tests is pulled out into its own function, and then we can do something like `if any([self.object_published(obj), self.ancestor_published(obj), self.valid_id0(obj)])` in `is_exportable`. That way, if you wanted to add or remove a test it's just either adding a method and then adding it to that list OR removing an item from the list of functions that are called.